lupa-core/src/main/kotlin/org/jetbrains/research/lupa/kotlinAnalysis/util/JupyterInverter.kt

```diff
@@ -0,0 +1,144 @@
+package org.jetbrains.research.lupa
+
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.File
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.*
+
+class JupyterInverter(input: String, output: String) {
+    val inputDir = input
+    val outputDir = output
+
+    class Notebook(notebookJSON: JSONObject, filename: String) {
+        var markdownCells: JSONArray
+        var codeCells: JSONArray
+        val metaData: JSONObject = notebookJSON.get("metadata") as JSONObject
+        val ntbName: String = filename
+
+        init {
+            this.markdownCells = getMarkdownCells(notebookJSON)
+            this.codeCells = getCodeCells(notebookJSON)
+        }
+
+        private fun cellToArray(cellJSON: JSONObject): JSONArray {
+            val source = cellJSON.get("source")
+            return if (source is String) {
+                val arr = source.split("\n").toMutableList()
+                for (i in 0 until arr.size - 1) {
+                    arr[i] = arr[i] + "\n"
+                }
+
+                JSONArray(arr)
+            } else source as JSONArray
+        }
+
+        private fun getCellLength(cellJSON: JSONObject): Int {
+            return cellToArray(cellJSON).length()
+        }
+
+        private fun getCellsByType(ntbJSON: JSONObject, cellType: String): JSONArray {
+            val resultedArray: ArrayList<JSONObject> = arrayListOf()
+            val cellArray: JSONArray = ntbJSON.get("cells") as JSONArray
```

Comment:
Please, avoid using string constants in the code. You can create an enum class for it, e.g.
```
 enum class NotebookJsonCell(val key: String) {
            SOURCE("source")
        }
```
and then
```
val source = cellJSON.get(NotebookJsonCell.SOURCE.key)
```